src/async-paginate.jsx

```diff
@@ -16,6 +16,13 @@ const sleep = (ms) => new Promise((resolve) => {
   }, ms);
 });
 
+// Supports forwardRef https://github.com/facebook/prop-types/issues/200
+const ComponentPropType = PropTypes.oneOfType([
+  PropTypes.func,
+  PropTypes.string,
+  PropTypes.shape({ render: PropTypes.func.isRequired }),
```

Comment:
Atlaskit is using styled-components that are using [forwardRef]() feature from React.

There are know issue with this approach combined with usage of `PropTypes`:
 - https://github.com/styled-components/styled-components/issues/2271#issuecomment-444411005
-  https://github.com/facebook/prop-types/issues/200